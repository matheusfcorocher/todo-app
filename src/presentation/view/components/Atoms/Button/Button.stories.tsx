import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import Button from './Button';

export default {
  title: 'Atoms/Button',
  component: Button,
  decorators: [
    (Story) => (
      <ul style={{ listStyleType: 'none', margin: 0, padding: 0 }}>
        <Story />
      </ul>
    ),
  ],
  argTypes: {
    variant: {
      options: ['primary', 'secondary', 'tertiary'],
      control: { type: 'inline-radio' }
    },
    size: {
      options: ["small" , "medium" , "large"],
      control: { type: 'inline-radio' }
    },
    fontColor: {
      options: ["white" , "gray" , "dark"],
      control: { type: 'inline-radio' }
    },
  }
} as ComponentMeta<typeof Button>;

const Template: ComponentStory<typeof Button> = (args) =>
  <Button {...args} handleFunction={function (): void {
  }} />;

export const Default = Template.bind({});
Default.args = {};

export const PrimaryButton = Template.bind({});
PrimaryButton.args = {
  title: "Click Me",
};

