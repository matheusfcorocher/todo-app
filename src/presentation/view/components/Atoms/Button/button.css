@import '../../../styles/spacing/spacing.css';
@import '../../../styles/colors/colors.css';
@import '../../../styles/typography/typography.css';
@import '../../../styles/round-corners/round-corners.css';

.icon-button {
    border: 0;
    font-size: var(--fs-300);
    font-family: var(--ff-serif);
    border-radius: var(--round-corner-sm);
}

.large {
    padding: var(--spacing-md);
}

.medium {
    padding: var(--spacing-sm);
}

.small {
    padding: var(--spacing-xs);
}

.primary {
    background-color: var(--primary-color);
}

.secondary {
    background-color: var(--secondary-color);
}

.tertiary {
    background-color: var(--tertiary-color);
}

.white {
    color: var(--white-color);
}

.gray {
    color: var(--gray-color);
}

.dark {
    color: var(--dark-color);
}

