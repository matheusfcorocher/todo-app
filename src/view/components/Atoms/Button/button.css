@import '../../../foundations/spacing/spacing.css';
@import '../../../foundations/colors/colors.css';
@import '../../../foundations/typography/typography.css';

.icon-button {
    border: 0;
    font-size: var(--fs-300);
    font-family: var(--ff-serif);
}

.large {
    padding: var(--spacing-md);
}

.medium {
    padding: var(--spacing-sm);
}

.small {
    padding: var(--spacing-xs);
}

.primary {
    background-color: var(--primary-color);
}

.secondary {
    background-color: var(--secondary-color);
}

.tertiary {
    background-color: var(--tertiary-color);
}

.white {
    color: var(--white-color);
}

.gray {
    color: var(--gray-color);
}

.dark {
    color: var(--dark-color);
}

