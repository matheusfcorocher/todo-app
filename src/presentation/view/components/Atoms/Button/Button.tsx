import './button.css';

interface ButtonProps {    
    handleFunction: () => void,
    title?: string,
    size?: "small" | "medium" | "large",
    variant?: "primary" | "secondary" | "tertiary",
    fontColor?: "white" | "gray" | "dark",
}

function Button({
    handleFunction,
    title,
    size="large",
    variant="primary",
    fontColor="dark"
}: ButtonProps) {
    return (
        <button
            title={title}
            className={`icon-button ${size} ${variant} ${fontColor}`}
            onClick={() => handleFunction()}
            role="button"
        >
            {title}
        </button>
    );
}

export default Button;
