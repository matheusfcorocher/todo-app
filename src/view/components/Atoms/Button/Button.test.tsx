import { render, screen, cleanup } from '@testing-library/react';
import Button from './Button';

describe("::Components ::Atoms ::Button", () => {
    test('renders Button', () => {
        render(<Button handleFunction={function (): void {
            throw new Error('Function not implemented.');
        }}/>);
        const icon = screen.getByRole("button");
        expect(icon).toMatchSnapshot();
    })
})
