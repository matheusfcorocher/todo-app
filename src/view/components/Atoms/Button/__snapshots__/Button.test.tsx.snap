// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`::Components ::Atoms ::Button renders Button 1`] = `
<button
  class="icon-button large primary-color"
  role="button"
/>
`;
